fairlearn/reductions/_moments/conditional_selection_rate.py

```diff
@@ -29,17 +33,38 @@ class ConditionalSelectionRate(ClassificationMoment):
 
     """
 
+    def __init__(self, ratio=1.0):
+        """Intialise with the ratio value."""
+        super(ConditionalSelectionRate, self).__init__()
+        if ratio <= 0 or ratio > 1:
+            raise ValueError(_MESSAGE_RATIO_NOT_IN_RANGE)
+        self.ratio = ratio
+
     def default_objective(self):
         """Return the default objective for moments of this kind."""
         return ErrorRate()
 
-    def load_data(self, X, y, event=None, **kwargs):
+    def load_data(self, X, y, event=None, utilities=None, **kwargs):
         """Load the specified data into this object.
 
         This adds a column `event` to the `tags` field.
+
+        The `utilities` is a matrix which correspond to g(X,A,Y,h(X)) as mentioned
+        in the paper `Agarwal et al. (2018) <https://arxiv.org/abs/1803.02453>`
+        The `utilities` defaults to h(X), i.e. [1, 0] for each X_i
+        The `_POSITIVE_UTILITY` is G^1 and the `_NEGATIVE_UTILITY` is G^0.
+        Assumes binary classification with labels 0/1.
```

Comment:
We should change the semantics, so the utility for 0 is in the first column and the utility for 1 is in the second column. This will make it consistent with the output of `predict_proba` for example.

Reply:
This is nice. I'll change `self.utilities` to have 2d-array-like format (similar to `predict_proba` format) as it definitely feels more intuitive. I'll do away with specific strings `_POSITIVE_UTILITY` and `NEGATIVE_UTILITY`.